### PR TITLE
Add acceleration management UI 

### PR DESCRIPTION
**Pull request by ps48**
### Description
Add acceleration management UI skeleton

### Issues Resolved
* Acc. Home Page 
* Selector for Data source
* Acceleration management Table 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ps48** [comment]
The skeleton is subject to change based on backend connection. Will have to add unit tests later after Backend connection is tested.

**Comment by ps48** [comment]
https://github.com/opensearch-project/dashboards-observability/assets/4348487/1e99f0c2-e176-4173-a5bc-b4bbe6e38f43

File: public/components/data_connections/components/acceleration_ui/acceleration_header.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import {
+  EuiLink,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import React from 'react';
+import { OPENSEARCH_DOCUMENTATION_URL } from '../../../../../common/constants/integrations';
```
Comment (derek-ho) [inline public/components/data_connections/components/acceleration_ui/acceleration_header.tsx:15]: Create a new one for datasources?

File: public/components/data_connections/components/datasource.tsx [hunk 2]
```diff
@@ -164,7 +164,11 @@ export function DataSource(props: any) {
                 icon={<EuiIcon size="xxl" type="bolt" />}
                 title={'Accelerate performance'}
                 description="Accelerate performance through OpenSearch indexing."
-                onClick={() => {}}
+                onClick={() =>
+                  window.location.assign(
+                    `${observabilityDatasourcesID}#/acceleration/${dataSource}`
```
Comment (derek-ho) [inline public/components/data_connections/components/datasource.tsx:169]: why do you need the datasourcesId here?

File: public/components/data_connections/components/acceleration_ui/acceleration_header.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        <EuiPageHeaderSection>
+          <EuiTitle size="l" data-test-subj="acceleration-header">
+            <h1>Acceleration Indices</h1>
+          </EuiTitle>
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <EuiSpacer size="s" />
+      <EuiText size="s" color="subdued">
+        Manage acceleration indices from external data connections.{' '}
+        <EuiLink external={true} href={OPENSEARCH_DOCUMENTATION_URL} target="blank">
```
Comment (joshuali925) [inline public/components/data_connections/components/acceleration_ui/acceleration_header.tsx:30]: i think we mostly use `_blank`, otherwise different links will replace previous page in the same new tab

File: public/components/data_connections/components/acceleration_ui/acceleration_indices.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        text: 'Acceleration',
+        href: '#/manage/acceleration/' + { dataSource },
+      },
+    ]);
+  }, [dataSource]);
+  return (
+    <EuiPage>
+      <EuiPageBody component="div">
+        <AccelerationHeader />
+        <EuiPageContent data-test-subj="manageAccelerationIndices">
```
Comment (joshuali925) [inline public/components/data_connections/components/acceleration_ui/acceleration_indices.tsx:33]: minor issue, some data-test-subj are kebab-case some are camelCase?

File: public/components/data_connections/components/acceleration_ui/management/management_table.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import {
+  EuiIcon,
+  EuiInMemoryTable,
+  EuiSpacer,
+  EuiTableFieldDataColumnType,
+  EuiText,
+} from '@elastic/eui';
+import _ from 'lodash';
+
+interface AccelrationIndicesRecordType {
```
Comment (joshuali925) [inline public/components/data_connections/components/acceleration_ui/management/management_table.tsx:16]: typo? same for `accelrationIndicesRecords, setAccelrationIndicesRecords`
```suggestion
interface AccelerationIndicesRecordType {
```

File: public/components/data_connections/components/acceleration_ui/management/management_table.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      render: (value, record) => (
+        <EuiIcon
+          type={'trash'}
+          onClick={() => {
+            /* Delete Datasource*/
+          }}
+        />
+      ),
+    },
+  ] as Array<EuiTableFieldDataColumnType<any>>;
```
Comment (joshuali925) [inline public/components/data_connections/components/acceleration_ui/management/management_table.tsx:76]: should it be 
```typescript
const tableColumns: Array<EuiTableFieldDataColumnType<AccelrationIndicesRecordType>> = [
```

File: public/components/data_connections/components/acceleration_ui/management/management_table.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    <>
+      <EuiSpacer size="s" />
+      <EuiInMemoryTable
+        items={accelrationIndicesRecords}
+        itemId="id"
+        columns={tableColumns}
+        tableLayout="auto"
+        pagination={{
+          initialPageSize: 10,
+          pageSizeOptions: [5, 10, 15],
```
Comment (joshuali925) [inline public/components/data_connections/components/acceleration_ui/management/management_table.tsx:107]: i know we have many tables with options 5, 10, 15, but maybe they are too small? in dashboards core many tables use 10, 20, 50 with default 20

File: public/components/data_connections/components/datasources_header.tsx [hunk 8]
```diff
@@ -15,7 +15,7 @@ import _ from 'lodash';
 import React from 'react';
 import { OPENSEARCH_DOCUMENTATION_URL } from '../../../../common/constants/data_connections';
 
-export function DataConnectionsHeader() {
+export const DataConnectionsHeader = () => {
```
Comment (joshuali925) [inline public/components/data_connections/components/datasources_header.tsx:18]: just curious why this change?

File: public/components/data_connections/components/acceleration_ui/acceleration_header.tsx (hunk 1 above)
Comment (ps48) [reply public/components/data_connections/components/acceleration_ui/acceleration_header.tsx]: done

File: public/components/data_connections/components/datasource.tsx (hunk 2 above)
Comment (ps48) [reply public/components/data_connections/components/datasource.tsx:169]: Yes it wil be. But window assign directly replaces the route parent `localhost:9200/app/datasources#/acc/..` to `localhost:9200/acc..` along with the suffix. Thus need to add the route parent here.

Comment (joshuali925) [reply public/components/data_connections/components/datasource.tsx:169]: you might want to navigate to url instead of explicit location.assign https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a6af77d6422a3b54fdcbaea7b19b5b2d541508b0/src/core/public/application/types.ts#L728-L764

File: public/components/data_connections/components/acceleration_ui/acceleration_header.tsx (hunk 3 above)
Comment (ps48) [reply public/components/data_connections/components/acceleration_ui/acceleration_header.tsx]: Thanks for the catch!

File: public/components/data_connections/components/acceleration_ui/management/management_table.tsx (hunk 5 above)
Comment (ps48) [reply public/components/data_connections/components/acceleration_ui/management/management_table.tsx]: Thanks for the catch!

File: public/components/data_connections/components/acceleration_ui/management/management_table.tsx (hunk 6 above)
Comment (ps48) [reply public/components/data_connections/components/acceleration_ui/management/management_table.tsx]: yes

File: public/components/data_connections/components/acceleration_ui/management/management_table.tsx (hunk 7 above)
Comment (ps48) [reply public/components/data_connections/components/acceleration_ui/management/management_table.tsx]: done!

File: public/components/data_connections/components/datasources_header.tsx (hunk 8 above)
Comment (ps48) [reply public/components/data_connections/components/datasources_header.tsx:18]: Wanted to make it consistent with rest of the observability.

File: public/components/data_connections/components/acceleration_ui/acceleration_indices.tsx (hunk 4 above)
Comment (ps48) [reply public/components/data_connections/components/acceleration_ui/acceleration_indices.tsx]: done

File: public/components/data_connections/components/datasource.tsx (hunk 2 above)
Comment (ps48) [reply public/components/data_connections/components/datasource.tsx:169]: @joshuali925 will need some help here. I tried to change code as below:
```
coreRefs.application?.navigateToUrl(`#/acceleration/${dataSource}`)
```
The URL shifts from `http://localhost:5601/app/datasources#/manage/my_spark1` to `http://localhost:5601/app/datasources#/acceleration/my_spark1` but doesn't load the acceleration page. I might be doing something wrong here.

Comment (joshuali925) [reply public/components/data_connections/components/datasource.tsx:169]: hm not sure, the code is going to call history.push https://github.com/opensearch-project/opensearch-dashboards/blob/0709333c641f1e2bcaf116eb7d5504b33afcd71d/src/core/public/application/application_service.tsx#L138

and it seems to be a known issue in react router https://stackoverflow.com/questions/42941708/react-history-push-is-updating-url-but-not-navigating-to-it-in-browser

i brought this up because it's consistent with core and handles some edge cases like cross app or cross site navigation, but if it doesn't work probably you can use location.assign or update location.hash

**Review by joshuali925 — APPROVED** [review]

File: public/components/data_connections/components/datasource.tsx (hunk 2 above)
Comment (ps48) [reply public/components/data_connections/components/datasource.tsx]: Resorted to use window.location.hash

**Review by derek-ho — APPROVED** [review]
